instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/instrumentation.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module GraphQL
+      # The Instrumentation class contains logic to detect and install the GraphQL
+      # instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        install do |config|
+          require_dependencies
+          patch_graphql(config)
+        end
+
+        present do
+          defined?(::GraphQL)
+        end
+
+        private
+
+        def require_dependencies
+          require_relative 'patches/opentelemetry_graphql_tracing'
+        end
+
+        def patch_graphql(config = {})
+          config[:schemas].each { |s| patch_schema!(s) }
+        end
+
+        def target_version
+          Integration.version
+        end
+
+        def patch_schema!(schema)
+          if schema.respond_to?(:use)
+            schema.use(
+              OpenTelemetry::Instrumentation::GraphQL::Patches::OpenTelemetryGraphQLTracing
+            )
+          else
+            schema.define do
+              use(
+                OpenTelemetry::Instrumentation::GraphQL::Patches::OpenTelemetryGraphQLTracing
+              )
+            end
+          end
+        # Protect our instrumentation from bad user input in schema config option
+        rescue StandardError => e
+          OpenTelemetry.logger.error("Unable to patch schema #{schema}: #{e.message}")
+          true
```

Comment:
I don't think this is necessary?

Reply:
i was concerned about users passing in a string, or a class which doesn't have the `define` method, etc. but if it's swallowed by broader error handling i can remove

Reply:
Sorry, I meant specifically returning `true`. The method isn't expected to return anything.